haskell-src/Concordium/Types.hs

```diff
@@ -529,6 +535,12 @@ instance HashableTo Hash.Hash MintRate where
 
 instance Monad m => MHashableTo m Hash.Hash MintRate
 
+instance Arbitrary MintRate where
+  arbitrary = do
+    mrMantissa <- arbitrary
+    mrExponent <- arbitrary
+    return MintRate{..}
```

Comment:
There are some unstated assumptions related to this type. The intention seems to be that the values are reasonable, and some functionality depends on this, see e.g. https://github.com/Concordium/concordium-base/issues/55

But here you are generating arbitrary values that are in general completely nonsensical for the mint rate.

Reply:
The idea behind (random) testing is exactly that: to uncover hidden assumptions and codify them. What does it mean for a `MintRate` value to be "reasonable"?

Reply:
I do not need generic lessons about randomized testing. I am well aware of it and its value and limitations. And in fact that is **exactly** how that issue that is linked was discovered.

I simply pointed out that what you are doing here contradicts some of the code that already exists. In particular the JSON parsing does not handle values as described in the linked issue.

In practice there would be no loss of generality if values were always less than 1.

Reply:
It would be much more constructive and to the point to state exactly that in your initial comment:

"The values of `MintRate` are expected to be less than 1."